ESP8266/ESP8266.h

```diff
@@ -371,6 +371,7 @@ class ESP8266 {
     // FW version specific settings and functionalities
     bool _tcp_passive;
     int32_t _recv_tcp_passive(int id, void *data, uint32_t amount, uint32_t timeout);
+    mbed::Callback<void()> _callback;
```

Comment:
This could be named as _data_available_cb, or something simlar. Just for the sake of clarity.